quesma/model/pipeline_aggregations/bucket_script.go

```diff
@@ -4,44 +4,85 @@ package pipeline_aggregations
 
 import (
 	"context"
+	"fmt"
 	"quesma/logger"
 	"quesma/model"
+	"quesma/util"
+	"strings"
 )
 
 type BucketScript struct {
-	ctx context.Context
+	*PipelineAggregation
+	script string
 }
 
-func NewBucketScript(ctx context.Context) BucketScript {
-	return BucketScript{ctx: ctx}
+func NewBucketScript(ctx context.Context, script string) BucketScript {
+	return BucketScript{script: script, PipelineAggregation: newPipelineAggregation(ctx, "_count")}
 }
 
 func (query BucketScript) AggregationType() model.AggregationType {
 	return model.PipelineMetricsAggregation // not sure
 }
 
 func (query BucketScript) TranslateSqlResponseToJson(rows []model.QueryResultRow) model.JsonMap {
-	if len(rows) == 0 {
-		logger.WarnWithCtx(query.ctx).Msg("no rows returned for bucket script aggregation")
-		return model.JsonMap{"value": 0}
-	}
-	var response []model.JsonMap
-	for _, row := range rows {
-		response = append(response, model.JsonMap{"value": row.Cols[0].Value})
-	}
-	return model.JsonMap{
-		"buckets": response,
+	const defaultValue = 0.
+	switch {
+	case query.script == "params.numerator != null && params.denominator != null && params.denominator != 0 ? params.numerator / params.denominator : 0":
```

Comment:
Ok, this does not look very generic and is kind of hacky. We can add it for now with some comments that we should implement this fully.